### PR TITLE
refactor(focus-lock): separate logic of FocusLock component

### LINKED ISSUE #3353: Separation of UI and Logic in the `FocusLock` Component
### Description

Improving code maintainability and readability.

### Problem Statement/Justification

Separate the logic in `FocusLock` component into a `use-focus-lock.ts` file.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

Please create a new branch from 2.0 branch for development.

### PR DESCRIPTION
**Pull request by hong-nguyen13**
Closes #3353 (Separation of UI and Logic in the FocusLock Component)

## Description

Separate the UI and logic of FocusLock component

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: .changeset/cuddly-boats-search.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/focus-lock": minor
```
Comment (kaehehehe) [inline .changeset/cuddly-boats-search.md:2]: Please change minor to patch.

**Review by kaehehehe — CHANGES_REQUESTED** [review]

File: .changeset/cuddly-boats-search.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/focus-lock": patch
```
Comment (kaehehehe) [inline .changeset/cuddly-boats-search.md:2]: Sorry! 😢 The changeset is not needed for this PR, as it is related to refactoring. Please delete this file. @hong-nguyen13

**Review by kaehehehe — CHANGES_REQUESTED** [review]

File: .changeset/cuddly-boats-search.md (hunk 2 above)
Comment (hong-nguyen13) [reply .changeset/cuddly-boats-search.md]: I've deleted it! Thank you👌

**Review by kaehehehe — CHANGES_REQUESTED** [review]
The PR should be merged into the 2.0 branch, not the main branch!

スクリーンショット 2024-11-08 134830

**Review by kaehehehe — COMMENTED** [review]
You don't need to merge the main branch. There are too many changed files.

**Comment by hong-nguyen13** [comment]
> You don't need to merge the main branch. There are too many changed files.

I've changed the main branch to v2.0, what should I do next?

**Review by kaehehehe — APPROVED** [review]

**Comment by hirotomoyamada** [comment]
@hong-nguyen13 

If this PR is incorporated first, conflicts will occur in other work, so close this PR.

Please submit this PR again after the other work is completed.

**Pull request closed without merging**